edx_rbac/utils.py

```diff
@@ -0,0 +1,135 @@
+        # URIs to other paths.
+        return RequestFactory(
+            SERVER_NAME=parsed_url.hostname,
+            SERVER_PORT=parsed_url.port or 80,
+        ).get("/")
+
+    else:
+        return request
+
+
+def get_decoded_jwt_from_request(request):
+    """
+    Grab jwt from request if possible.
+
+    Returns a decoded jwt dict if it finds it.
+    Returns a None if it does not.
+    """
+    jwt_cookie = request.COOKIES.get(jwt_cookie_name(), None)
+
+    if not jwt_cookie:
+        return None
+    return jwt_decode_handler(jwt_cookie)
+
+
+def request_user_has_implicit_access_via_jwt(decoded_jwt, role_name):
+    """
+    Check the request's user access by mapping user's roles found in jwt to local feature roles.
+
+    decoded_jwt is a dict
+    role_name is a string
+
+    Returns a boolean.
+
+    Mapping should be in settings and look like:
+
+        SYSTEM_TO_FEATURE_ROLE_MAPPING = {
+            'enterprise_admin': ['coupon-management', 'data_api_access'],
+            'enterprise_leaner': [],
+            'coupon-manager': ['coupon-management']
+        }
+    """
+    jwt_roles_claim = decoded_jwt.get('roles', [])
+
+    feature_roles = []
+    for role_data in jwt_roles_claim:
+        role_in_jwt = role_data.split(':')[0]  # split should be more robust because of our cousekeys having colons
+        mapped_roles = settings.SYSTEM_TO_FEATURE_ROLE_MAPPING.get(role_in_jwt, [])
+        feature_roles.extend(mapped_roles)
+
+    if role_name in feature_roles:
+        return True
+    return False
+
+
+def user_has_access_via_database(user, role_name, role_assignment_class):
+    """
+    Check if there is a role assignment for a given user and role.
+
+    The role object itself is found via the role_name
+    """
+    try:
+        role_assignment_class.objects.get(user=user, role__name=role_name)
+    except role_assignment_class.DoesNotExist:
+        return False
+    return True
+
+
+def create_role_auth_claim_for_user(user):
+    """
+    Create role auth claim for a given user.
+
+    Takes a user, and for each RoleAssignment class specified in config as a
+    system wide jwt role associated with that user, creates a list of strings
+    denoting the role and context.
+
+    Returns a list.
+
+    This setting is a list of classes whose roles should be added to the
+    jwt. The setting should look something like this:
+
+        SYSTEM_WIDE_ROLE_CLASSES = [
+            SystemWideConcreteUserRoleAssignment
+        ]
+    """
+    role_auth_claim = []
+    for system_role_class in settings.SYSTEM_WIDE_ROLE_CLASSES:
+        app_name, model_name = system_role_class.split('.')
+        model_class = apps.get_model(app_name, model_name)
+
+        role_assignments = model_class.objects.filter(
+            user=user
+        ).select_related('role')
+
+        for role_assignment in role_assignments:
+            role_string = '{role}:{context}'.format(
```

Comment:
i think we should add some logic here to check if the context is set to something, because not all roles will map to a context

Reply:
ah right. in the case of no context, just a role, and no colon?

Reply:
yep